docs/internals/protocol/dataformats.rst

```diff
@@ -53,6 +53,40 @@ Note: zero-length arrays (and sets) are represented as a 12-byte value where
 ``dims`` equal to zero regardless of the shape in type descriptor.
 
 
+Sets of arrays are a special case and are represented as the following
+structure:
+
+.. code-block:: c
+
+    struct SetOfArrayValue {
+        // Number of dimensions, currently must
+        // always be 0 or 1.
```

Comment:
Maybe clarify what dimensions 0 and 1 mean?

Reply:
I'm not sure what I would say besides `0 means 0 dimensions and 1 means 1 dimension.` Maybe I don't fully understand their significance.

Reply:
@1st1 is there something specific you were thinking here?

Reply:
I'd clarify that zero dimensions is just an empty array.

Reply:
What @elprans said :)

Reply:
changed